.github/actions/docker-pull-image/action.yml

```diff
@@ -0,0 +1,126 @@
+name: 'Pull Docker Image'
+description: 'Login to a registry and pull a tagged image into the local cache'
+inputs:
+  image_tag:
+    description: 'Local image tag (e.g., nightly-vllm-amd64)'
+    required: true
+  repository:
+    description: 'Repository path (default ai-dynamo/dynamo)'
+    required: false
+    default: 'ai-dynamo/dynamo'
+  provider:
+    description: 'Registry provider: azure | aws'
+    required: false
+    default: 'azure'
+  # Azure inputs
+  azure_acr_hostname:
+    description: 'Azure ACR hostname'
+    required: false
+  azure_acr_user:
+    description: 'Azure ACR username'
+    required: false
+  azure_acr_password:
+    description: 'Azure ACR password'
+    required: false
+  # AWS inputs
+  aws_account_id:
+    description: 'AWS Account ID'
+    required: false
+  aws_default_region:
+    description: 'AWS Default Region'
+    required: false
+  aws_access_key_id:
+    description: 'AWS Access Key ID'
+    required: false
+  aws_secret_access_key:
+    description: 'AWS Secret Access Key'
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Login to Azure ACR
+      if: ${{ !(inputs.provider == 'aws' || env.PREFER_PROVIDER == 'aws' || inputs.aws_account_id != '') }}
+      shell: bash
+      env:
+        AZURE_ACR_HOSTNAME: ${{ inputs.azure_acr_hostname || env.AZURE_ACR_HOSTNAME }}
+      run: |
+        retry=0
+        until [ $retry -ge 3 ]; do
```

Comment:
I don't think we should implement retries unless there is proof of transient errors that have occurred. Lets simplify all of our logic and remove retries.

If we implement retries and errors are not transient, we are sending pointless API requests that can result in rate-limiting.